schema/schema.go

```diff
@@ -346,6 +347,55 @@ func (schema *Schema) StateVersioning() bool {
 	return stateful
 }
 
+func (schema *Schema) SyncKeyTemplate() string {
+	syncKeyTemplateRaw, ok := schema.Metadata["sync_key_template"]
+	if !ok {
+		return ""
+	}
+	syncKeyTemplate, ok := syncKeyTemplateRaw.(string)
+	if !ok {
+		return ""
+	}
+	return syncKeyTemplate
+}
+
+func (schema *Schema) GenerateCustomPath(data map[string]interface{}) (string, error) {
```

Comment:
let's use pongo2 template instead of defining function by ourself.
https://github.com/flosch/pongo2

pongo2 is already used in extension.